### PR TITLE
[1.2.x] Add markdown doc for annotation

### LINKED ISSUE #22262: Markdown Documentation null for AfterEach annotation
**Description:**
$title

**Affected Versions:**
v1.2.0 at least

### PR DESCRIPTION
**Pull request by KavinduZoysa**
## Purpose
Add markdown documentation attachment for annotation

Fixes #22262 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java [hunk 1]
```diff
@@ -509,7 +509,7 @@ public String toString() {
      *
      * @since 0.995.0
      */
-    public static class BIRAnnotation extends BIRNode {
+    public static class BIRAnnotation extends BIRDocumentableNode {
```
Comment (MaryamZi) [inline compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java:512]: @hasithaa we can go ahead with this change for the patch branch right?

File: tests/jballerina-unit-test/src/test/resources/test-src/balo/test_projects/test_documentation/src/test/test_documentation_symbol.bal [hunk 2]
```diff
@@ -25,3 +25,16 @@ public type Person object {
...
+# + b - 'field b' documentation
+# + c - 'field c' documentation
+type Tst record {
+    string a = "";
+    string b = "";
+    string c = "";
+};
+
+# Documentation for Test annotation
+annotation Tst Test on function;
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/balo/test_projects/test_documentation/src/test/test_documentation_symbol.bal:40]: Shouldn't this be public?

File: tests/jballerina-unit-test/src/test/resources/test-src/balo/test_balo/documentation/test_documentation.bal [hunk 3]
```diff
@@ -4,3 +4,7 @@ import testDocOrg/test;
 function testDocDummy() returns (boolean) {
     return test:open("testt");
 }
+
+@test:Test
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/resources/test-src/balo/test_balo/documentation/test_documentation.bal:8]: According to the annotation definition this should have `{}` right? So this probably fails at compilation? 

Shall we assert that there are no compilation errors?

File: tests/jballerina-unit-test/src/test/resources/test-src/balo/test_projects/test_documentation/src/test/test_documentation_symbol.bal (hunk 2 above)
Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/resources/test-src/balo/test_projects/test_documentation/src/test/test_documentation_symbol.bal]: Changed it

File: tests/jballerina-unit-test/src/test/resources/test-src/balo/test_balo/documentation/test_documentation.bal (hunk 3 above)
Comment (KavinduZoysa) [reply tests/jballerina-unit-test/src/test/resources/test-src/balo/test_balo/documentation/test_documentation.bal]: Yes, assert the compile errors

File: compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java (hunk 1 above)
Comment (MaryamZi) [reply compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BIRNode.java:512]: Discussed offline and decided this was OK.

File: tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/documentation/DocumentationTest.java [hunk 4]
```diff
@@ -104,6 +105,21 @@ public void testDocAttachmentBalo() {
                 .replaceAll(CARRIAGE_RETURN_CHAR, EMPTY_STRING), "True if male");
     }
 
+    @Test(description = "Test doc attachments in annotations")
+    public void testAnnotationDoc() {
+        PackageNode packageNode = result.getAST();
+        Assert.assertEquals(result.getErrorCount(), 0);
```
Comment (MaryamZi) [inline tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/balo/documentation/DocumentationTest.java:111]: Let's move this assertion to the `setup` method in a separate PR.

**Pull request merged**